### PR TITLE
superagent: req.responseType, res.xhr

### PR DESCRIPTION
**Pull request by benmosher**
Added:
- `SuperAgentRequest.responseType` setter method
- `SuperAgentResponse.xhr` field definition

Source reference from tests at 2.0.0 tag: https://github.com/visionmedia/superagent/blob/v2.0.0/test/client/request.js#L110 (included URL in superagent-tests.ts)

Since this test existed at 2.0.0 tag, I believe this could be published as the next patch version of the 2.0.x typing.

Needs review by a DefinitelyTyped member, per PR template.

### CODE REVIEW

File: superagent/superagent-tests.ts [hunk 1]
```diff
@@ -302,4 +303,13 @@ request
...
\ No newline at end of file
+  .catch((error) => {});
+
+// Requesting binary data.
+// adapted from: https://github.com/visionmedia/superagent/blob/v2.0.0/test/client/request.js#L110
+request
+  .get('/blob')
+  .responseType('blob')
+  .end(function(err, res){
+    assert.deepEqual(res.xhr.response instanceof Blob, true);
```
Comment (zhengbli) [inline superagent/superagent-tests.ts]: Can you just do `assert(res.xhr.response instanceof Blob)`

Comment (zhengbli) [inline superagent/superagent-tests.ts]: Maybe add `assert(res.xhr instanceof XMLHttpRequest)` as well

Comment (benmosher) [inline superagent/superagent-tests.ts]: Can do. I was just trying to keep it as close to the original test source as possible.

File: superagent/superagent.d.ts [hunk 2]
```diff
@@ -77,6 +77,7 @@ declare module "superagent" {
       notAcceptable: boolean;
       notFound: boolean;
       forbidden: boolean;
+      xhr: XMLHttpRequest;
```
Comment (denneulin) [inline superagent/superagent.d.ts]: I get a TS2304 error `Cannot find name 'XMLHttpRequest'` because of line 80 `xhr: XMLHttpRequest;`

Comment (zhengbli) [inline superagent/superagent.d.ts]: `XMLHttpRequest` exists in lib.d.ts though. Can you provide a minimum repro sample code and open a separate issue in this repo for better tracking? Thanks!

**Review by zhengbli — COMMENTED** [review]
Not actually a code review

File: superagent/superagent.d.ts (hunk 2 above)
Comment (denneulin) [inline superagent/superagent.d.ts]: Thank you for your answer. 
You helped me to point my error => "dom" was missing in my tsconfig lib entry.

**Pull request merged**